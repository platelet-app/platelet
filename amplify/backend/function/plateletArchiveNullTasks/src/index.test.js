const indexModule = require("./index");
const { default: fetch } = require("node-fetch");

const mockTaskNewer = {
    id: "anotherTaskId",
    archived: null,
    _version: 1,
    _deleted: null,
    _lastChangedAt: 1620000000000,
};

const mockTask = {
    id: "someTaskId",
    archived: null,
    _version: 1,
};

const fakeData = {
    listTasks: {
        items: [mockTask],
        nextToken: "tasktoken",
    },
};
const fakeDataSecond = {
    listTasks: {
        items: [mockTaskNewer],
        nextToken: null,
    },
};

const fakeLocationData = {
    listLocations: {
        items: [
            {
                id: "someLocationId",
                archived: null,
                _version: 1,
            },
        ],
        nextToken: "loctoken",
    },
};

const fakeLocationDataSecond = {
    ...fakeLocationData,
    listLocations: {
        ...fakeLocationData.listLocations,
        nextToken: null,
    },
};

const fakeAssigneeData = {
    listTaskAssignees: {
        items: [
            {
                id: "someAssignmentId",
                archived: null,
                _version: 1,
            },
        ],
        nextToken: "asstoken",
    },
};

const fakeAssigneeDataSecond = {
    ...fakeAssigneeData,
    listTaskAssignees: {
        ...fakeAssigneeData.listTaskAssignees,
        nextToken: null,
    },
};

const fakeAssigneeReturn = {
    updateTaskAssignee: {
        id: "someAssignmentId",
        archived: 0,
    },
};
const fakeDeliverableData = {
    listDeliverables: {
        items: [
            {
                id: "someDeliverableId",
                archived: null,
                _version: 1,
            },
        ],
        nextToken: "deltoken",
    },
};
const fakeDeliverableDataSecond = {
    ...fakeDeliverableData,
    listDeliverables: {
        ...fakeDeliverableData.listDeliverables,
        nextToken: null,
    },
};
const fakeDeliverableReturn = {
    updateDeliverable: {
        id: "someDeliverableId",
        archived: 0,
    },
};
const fakeCommentData = {
    listComments: {
        items: [
            {
                id: "someCommentId",
                _version: 1,
                archived: null,
            },
        ],
        nextToken: "comtoken",
    },
};

const fakeCommentDataSecond = {
    ...fakeCommentData,
    listComments: {
        ...fakeCommentData.listComments,
        nextToken: null,
    },
};

const fakeCommentReturn = {
    updateComment: {
        id: "someCommentId",
        archived: 0,
    },
};
const fakeTaskReturn = {
    updateTask: {
        id: "someTaskId",
        archived: 0,
    },
};

const fakeLocationReturn = {
    updateLocation: {
        id: "someLocationId",
        archived: 0,
    },
};

function setupFetchStub(data) {
    return function fetchStub(_url) {
        return new Promise((resolve) => {
            resolve({
                json: () =>
                    Promise.resolve({
                        data,
                    }),
            });
        });
    };
}

describe("plateletArchiver", () => {
    afterEach(() => jest.restoreAllMocks());

    it("should return a function", () => {
        expect(typeof indexModule.handler).toBe("function");
    });

    test("convert items from null archived to 0", async () => {
        const requestSpy = jest.spyOn(indexModule, "makeNewRequest");
        jest.spyOn(fetch, "default")
            .mockImplementationOnce(setupFetchStub(fakeData))
            .mockImplementationOnce(setupFetchStub(fakeTaskReturn))
            .mockImplementationOnce(setupFetchStub(fakeCommentData))
            .mockImplementationOnce(setupFetchStub(fakeCommentReturn))
            .mockImplementationOnce(setupFetchStub(fakeLocationData))
            .mockImplementationOnce(setupFetchStub(fakeLocationReturn))
            .mockImplementationOnce(setupFetchStub(fakeDeliverableData))
            .mockImplementationOnce(setupFetchStub(fakeDeliverableReturn))
            .mockImplementationOnce(setupFetchStub(fakeAssigneeData))
            .mockImplementationOnce(setupFetchStub(fakeAssigneeReturn));
        await indexModule.handler({}, indexModule.makeNewRequest);
        expect(requestSpy).toMatchSnapshot();
    });
    test.skip("with nextToken convert items from null archived to 0", async () => {
        const requestSpy = jest.spyOn(indexModule, "makeNewRequest");
        jest.spyOn(fetch, "default")
            .mockImplementationOnce(setupFetchStub(fakeData))
            .mockImplementationOnce(setupFetchStub(fakeDataSecond))
            .mockImplementationOnce(setupFetchStub(fakeTaskReturn))
            .mockImplementationOnce(setupFetchStub(fakeTaskReturn))
            .mockImplementationOnce(setupFetchStub(fakeCommentData))
            .mockImplementationOnce(setupFetchStub(fakeCommentDataSecond))
            .mockImplementationOnce(setupFetchStub(fakeCommentReturn))
            .mockImplementationOnce(setupFetchStub(fakeCommentReturn))
            .mockImplementationOnce(setupFetchStub(fakeLocationData))
            .mockImplementationOnce(setupFetchStub(fakeLocationDataSecond))
            .mockImplementationOnce(setupFetchStub(fakeLocationReturn))
            .mockImplementationOnce(setupFetchStub(fakeLocationReturn))
            .mockImplementationOnce(setupFetchStub(fakeDeliverableData))
            .mockImplementationOnce(setupFetchStub(fakeDeliverableDataSecond))
            .mockImplementationOnce(setupFetchStub(fakeDeliverableReturn))
            .mockImplementationOnce(setupFetchStub(fakeDeliverableReturn))
            .mockImplementationOnce(setupFetchStub(fakeAssigneeData))
            .mockImplementationOnce(setupFetchStub(fakeAssigneeDataSecond))
            .mockImplementationOnce(setupFetchStub(fakeAssigneeReturn))
            .mockImplementationOnce(setupFetchStub(fakeAssigneeReturn));
        await indexModule.handler({}, indexModule.makeNewRequest);
        expect(requestSpy).toMatchSnapshot();
    });
});
