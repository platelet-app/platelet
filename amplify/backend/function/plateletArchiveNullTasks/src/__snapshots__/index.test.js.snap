// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plateletArchiver convert items from null archived to 0 1`] = `
[MockFunction] {
  "calls": [
    [
      "
    query listTasks($nextToken: String) {
        listTasks(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": null,
      },
    ],
    [
      "
    query listTasks($nextToken: String) {
        listTasks(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": "tasktoken",
      },
    ],
    [
      "
    query listComments($nextToken: String) {
        listComments(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": null,
      },
    ],
    [
      "
    query listComments($nextToken: String) {
        listComments(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": "comtoken",
      },
    ],
    [
      "
    query listLocations($nextToken: String) {
        listLocations(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": null,
      },
    ],
    [
      "
    query listLocations($nextToken: String) {
        listLocations(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": "loctoken",
      },
    ],
    [
      "
    query listDeliverables($nextToken: String) {
        listDeliverables(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": null,
      },
    ],
    [
      "
    query listDeliverables($nextToken: String) {
        listDeliverables(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": "deltoken",
      },
    ],
    [
      "
    query listTaskAssignees($nextToken: String) {
        listTaskAssignees(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": null,
      },
    ],
    [
      "
    query listTaskAssignees($nextToken: String) {
        listTaskAssignees(nextToken: $nextToken) {
            items {
                id
                _version
                archived
            }
            nextToken
        }
    }
",
      {
        "nextToken": "asstoken",
      },
    ],
    [
      "
    mutation UpdateTask($input: UpdateTaskInput!) {
        updateTask(input: $input) {
            id
            archived
        }
    }
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someTaskId",
        },
      },
    ],
    [
      "
    mutation UpdateTask($input: UpdateTaskInput!) {
        updateTask(input: $input) {
            id
            archived
        }
    }
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "anotherTaskId",
        },
      },
    ],
    [
      "
    mutation UpdateComment($input: UpdateCommentInput!) {
        updateComment(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someCommentId",
        },
      },
    ],
    [
      "
    mutation UpdateComment($input: UpdateCommentInput!) {
        updateComment(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someCommentId",
        },
      },
    ],
    [
      "
    mutation UpdateLocation($input: UpdateLocationInput!) {
        updateLocation(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someLocationId",
        },
      },
    ],
    [
      "
    mutation UpdateLocation($input: UpdateLocationInput!) {
        updateLocation(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someLocationId",
        },
      },
    ],
    [
      "
    mutation UpdateDeliverable($input: UpdateDeliverableInput!) {
        updateDeliverable(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someDeliverableId",
        },
      },
    ],
    [
      "
    mutation UpdateDeliverable($input: UpdateDeliverableInput!) {
        updateDeliverable(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someDeliverableId",
        },
      },
    ],
    [
      "
    mutation UpdateTaskAssignee($input: UpdateTaskAssigneeInput!) {
        updateTaskAssignee(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someAssignmentId",
        },
      },
    ],
    [
      "
    mutation UpdateTaskAssignee($input: UpdateTaskAssigneeInput!) {
        updateTaskAssignee(input: $input) {
            id
            archived
        };
    };
",
      {
        "input": {
          "_version": 1,
          "archived": 0,
          "id": "someAssignmentId",
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
